test/unit/DPNSClient.spec.js

```diff
@@ -0,0 +1,156 @@
+const rewiremock = require('rewiremock/node');
+const Document = require('@dashevo/dpp/lib/document/Document');
+const createDapiClientMock = require('../../lib/test/mocks/createDapiClientMock');
+const dpnsDocumentFixture = require('../../lib/test/fixtures/getDpnsDocumentFixture');
+const createWalletMock = require('../../lib/test/mocks/createWalletMock');
+const InvalidArgumentError = require('../../lib/errors/InvalidArgumentError');
+
+describe('DPNSClient', () => {
+  let dapiClientMock;
+  let parentDocument;
+  let walletMock;
+  let DPNSClient;
+  let bUser;
+
+  let registerMethodMock;
+  let resolveMethodMock;
+  let resolveByRecordMethodMock;
+  let searchMethodMock;
+
+  let dpnsClient;
+
+  beforeEach(function beforeEach() {
+    dapiClientMock = createDapiClientMock(this.sinon);
+
+    parentDocument = dpnsDocumentFixture.getParentDocumentFixture();
+
+    dapiClientMock.fetchDocuments.resolves([parentDocument.toJSON()]);
+    dapiClientMock.getLastUserStateTransitionHash.resolves('562a795654476351646e34744d576866637979455673317a7476744b3676797a');
+
+    walletMock = createWalletMock(this.sinon);
+    walletMock.getAccount.returns({
+      getAddress: this.sinon.stub().returns({ address: 'address' }),
+      getPrivateKeys: this.sinon.stub().returns(['privateKey']),
+    });
+
+    registerMethodMock = this.sinon.stub().resolves(parentDocument);
+    resolveMethodMock = this.sinon.stub().resolves(parentDocument);
+    resolveByRecordMethodMock = this.sinon.stub().resolves(parentDocument);
+    searchMethodMock = this.sinon.stub().resolves([parentDocument]);
+
+    DPNSClient = rewiremock.proxy('../../lib/DPNSClient', {
```

Comment:
Let's just overwrite private properties here. We should avoid of using rewiremock as much as possible